File: python/mirascope/llm/content/text.py
"""The `Text` content class."""

from dataclasses import dataclass
from typing import Literal


@dataclass(kw_only=True)
class Text:
    """Text content for a message."""

    type: Literal["text"] = "text"

    text: str
    """The text content."""


@dataclass(kw_only=True)
class TextStartChunk:
    """Represents the start of a text chunk stream."""

    type: Literal["text_start_chunk"] = "text_start_chunk"

    content_type: Literal["text"] = "text"
    """The type of content reconstructed by this chunk."""


@dataclass(kw_only=True)
class TextChunk:
    """Represents an incremental text chunk in a stream."""

    type: Literal["text_chunk"] = "text_chunk"

    content_type: Literal["text"] = "text"
    """The type of content reconstructed by this chunk."""

    delta: str
    """The incremental text added in this chunk."""


@dataclass(kw_only=True)
class TextEndChunk:
    """Represents the end of a text chunk stream."""

    type: Literal["text_end_chunk"] = "text_end_chunk"

    content_type: Literal["text"] = "text"
    """The type of content reconstructed by this chunk."""

File: typescript/src/llm/content/text.ts
/**
 * Text content for a message.
 *
 * Represents plain text content in a message. This is the most common
 * content type for both user and assistant messages.
 */
export type Text = {
  readonly type: 'text';

  /** The text content. */
  readonly text: string;
};

/**
 * Signals the start of a text content block in the stream.
 */
export type TextStartChunk = {
  readonly type: 'text_start_chunk';
  readonly contentType: 'text';
};

/**
 * Contains incremental text content.
 */
export type TextChunk = {
  readonly type: 'text_chunk';
  readonly contentType: 'text';
  /** The incremental text added in this chunk. */
  readonly delta: string;
};

/**
 * Signals the end of a text content block in the stream.
 */
export type TextEndChunk = {
  readonly type: 'text_end_chunk';
  readonly contentType: 'text';
};

/** Create a TextStartChunk */
export function textStart(): TextStartChunk {
  return { type: 'text_start_chunk', contentType: 'text' };
}

/** Create a TextChunk */
export function textChunk(delta: string): TextChunk {
  return { type: 'text_chunk', contentType: 'text', delta };
}

/** Create a TextEndChunk */
export function textEnd(): TextEndChunk {
  return { type: 'text_end_chunk', contentType: 'text' };
}
